clients/fides-js/src/lib/initialize.ts

```diff
@@ -269,97 +269,109 @@ export const initialize = async ({
     debug?: boolean
   ) => Promise<FidesCookie>;
 } & FidesConfig): Promise<Partial<Fides>> => {
-  let shouldInitOverlay: boolean = options.isOverlayEnabled;
   let effectiveExperience = experience;
-  let fidesRegionString: string | null = null;
 
-  if (shouldInitOverlay) {
-    if (!validateOptions(options)) {
-      debugLog(
-        options.debug,
-        "Invalid overlay options. Skipping overlay initialization.",
-        options
-      );
-      shouldInitOverlay = false;
-    }
+  const fidesObj: Partial<Fides> = {
+    consent: cookie.consent,
+    fides_meta: cookie.fides_meta,
+    identity: cookie.identity,
+    fides_string: cookie.fides_string,
+    tcf_consent: cookie.tcf_consent,
+    experience,
+    geolocation,
+    options,
+    initialized: true,
+  };
 
-    fidesRegionString = await retrieveEffectiveRegionString(
-      geolocation,
+  if (!options.isOverlayEnabled) {
+    return fidesObj;
+  }
+
+  if (!validateOptions(options)) {
+    debugLog(
+      options.debug,
+      "Invalid overlay options. Skipping overlay initialization.",
       options
     );
+    return fidesObj;
+  }
 
-    let fetchedClientSideExperience = false;
+  const fidesRegionString = await retrieveEffectiveRegionString(
+    geolocation,
+    options
+  );
+  if (!fidesRegionString) {
+    debugLog(
+      options.debug,
+      `User location could not be obtained. Skipping overlay initialization.`
+    );
+    return fidesObj;
+  }
+  const needsFetch = !isPrivacyExperience(effectiveExperience);
+  if (needsFetch) {
+    // If no effective PrivacyExperience was pre-fetched, fetch one now from
+    // the Fides API using the current region string
+    effectiveExperience = await fetchExperience(
+      fidesRegionString,
+      options.fidesApiUrl,
+      options.debug,
+      cookie.identity.fides_user_device_id
+    );
+  }
 
-    if (!fidesRegionString) {
+  if (
+    !isPrivacyExperience(effectiveExperience) ||
+    !experienceIsValid(effectiveExperience, options)
+  ) {
+    debugLog(options.debug, `Experience is invalid ${effectiveExperience}`);
+    return fidesObj;
+  }
+
+  // At this point, we have a valid privacy experience
+  if (options.fidesString) {
```

Comment:
this part is still messy—I think the crux of what we want is to figure out "are we using experience or cookie as source of truth here?" 

1. if experience, call `updateCookie`
2. if cookie, call `updateExperience` 

right now, I _think_ the only reason we'd want to update the experience based on the cookie at this point (2️⃣) is if we have a passed in `fidesString`. but it seems like there may be more cases, and this is separate from the logic in `fides-tcf.ts` which initializes a prefetched experience...